plugins/lookup/hashi_vault.py

```diff
@@ -111,7 +114,12 @@
       aliases: [ as ]
       version_added: '0.2.0'
     mount_point:
-      description: Vault mount point, only required if you have a custom mount point.
+      description: Vault mount point, only required if you have a custom mount point. Does not apply to token authentication.
+    jwt:
+      description: The JSON Web Token (JWT) to use for JWT/OIDC authentication to Vault.
+      env:
+        - name: VAULT_JWT
```

Comment:
I think I'd like us to start switching to using `ANSIBLE_VAULT_` as the prefix for env vars going forward, to avoid ambiguities between what's Ansible-specific vs. env vars that are recognized by Vault itself: https://www.vaultproject.io/docs/commands#environment-variables

Reply:
A (potential) problem with `ANSIBLE_VAULT_` is that people might think they are related to [Ansible Vault](https://docs.ansible.com/ansible/latest/user_guide/vault.html). `ANSIBLE_HASHI_VAULT_` would be better, but it's a bit long.

Reply:
I think this is a good name as it is. 😄 The other environment variables already uses the same naming, and consistency is good. Even if this is not a natively supported environment variable in Vault CLI, I think it could have been. See this as a reference: https://github.com/hashicorp/vault-plugin-auth-kubernetes/issues/26

Reply:
Felix brings up a good point about Ansible Vault. 

It's true that `ANSIBLE_HASHI_VAULT_` is better, and also long. But there are other longer vars so it may not be too bad.

Erik, I know we use several with `VAULT_` already (and in a recent PR added some with `ANSIBLE_VAULT_`, maybe those will be changing too). But I'm looking to try to correct some past mistakes, and avoid what might become clashes in the future. 

Not saying it's all fully thought out but I want to give careful consideration to it. 
Leave it as is for the purposes of continuing to iterate on this PR, hopefully we'll be able to come up with something reasonable before it's ready to merge?

Reply:
Among your two suggestions, I'll vote for `ANSIBLE_HASHI_VAULT_` as the environment variable prefix. But I still think just using `VAULT_` is better. We are now using a combination of Ansible (with hashi_vault lookup) and Vault CLI in our CI/CD pipelines - using environment variables for the Vault client configuration. Some pipeline using Vault CLI, and some using Ansible. And I think it would be sad if "the same" environment variables were named differently. I do not think is confusing or misleading at all, but better. Vault CLI is one client, and Ansible hashi_vault (building on hvac) is another client. And the environment variables could, and in my opinion should, be named accordingly.

Reply:
Thanks @erikgb , to clarify, where it makes sense to bring in the value of an existing `VAULT_` env var, I want to to continue to do so! However, I want to make a distinction between values that have meaning only in ansible vs. those that have meaning outside of ansible too, and to apply what I think (hope) is intuitive precedence to them all. I've created an issue to track this as it's larger than this PR. I hope you'll take a look and help continue the conversation there: #1271

Reply:
@briantist I do not have a lot to contribute with in that discussion. We are only running Ansible from CI/CD jobs ( Ansible controller running on container images), and can cope with any naming of environment variables. But personally I prefer:
- To allow for optional environment variable configuration
- Not too long names on environment variables
- Avoid potentially misleading naming; the `ANSIBLE_VAULT_` prefix is a good example of that

Reply:
Ok, let's go with `ANSIBLE_HASHI_VAULT_JWT` for this, as I expect we'll be making moves to change existing vars where possible too, and using that prefix for new ones.

Reply:
@briantist Are you sure? 😉 I do not agree, but I am going to change it - if the change is a requirement to get your approval. 😄

Reply:
I think so, yes! The only negative to that name is length, and while I think shorter names are nice to have, I think it's less important than clarity. If you have more to say on the general direction let's take it to #1271. For your points:

* ✔Allows optional env var config
* ➖"too long" is subjective, but it's not too long from a technical limit point of view
* ✔This one is unambiguous. `ANSIBLE_VAULT_` unfortunately sounds too much like it's related to "Ansible Vault", and is in fact already in use for several "Ansible Vault"-specific settings: https://docs.ansible.com/ansible/latest/reference_appendices/config.html#default-vault-encrypt-identity

Reply:
Done